cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/DefaultArtifactManager.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.internal.app.runtime.artifact;
+
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.artifact.ArtifactInfo;
+import co.cask.cdap.api.artifact.CloseableClassLoader;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.lang.DirectoryClassLoader;
+import co.cask.cdap.common.lang.jar.BundleJarUtil;
+import co.cask.cdap.common.utils.DirUtils;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.security.spi.authentication.AuthenticationContext;
+import co.cask.cdap.security.spi.authorization.AuthorizationEnforcer;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Artifact store read only
+ */
+public class DefaultArtifactManager {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultArtifactManager.class);
+
+  private final ArtifactStore artifactStore;
+  private final AuthorizationEnforcer authorizationEnforcer;
+  private final AuthenticationContext authenticationContext;
+  private final File tmpDir;
+  private final ClassLoader bootstrapClassLoader;
+
+  @VisibleForTesting
+  @Inject
+  public DefaultArtifactManager(CConfiguration cConf,
+                                ArtifactStore artifactStore, AuthorizationEnforcer authorizationEnforcer,
+                                AuthenticationContext authenticationContext) {
+    this.artifactStore = artifactStore;
```

Comment:
This won't work. In distributed mode, user containers cannot use ArtifactStore directly as it won't have the permission to scan the HBase table in secure mode.